src/main/java/org/reaktivity/nukleus/http2/internal/Http2WriteScheduler.java

```diff
@@ -138,25 +138,32 @@ public boolean goaway(int lastStreamId, Http2ErrorCode errorCode)
     @Override
     public boolean rst(int streamId, Http2ErrorCode errorCode)
     {
-        long traceId = connection.factory.supplyTrace.getAsLong();
-        int length = 4;                     // 4 for RST_STREAM payload
-        int sizeof = length + 9;            // +9 for HTTP2 framing
         Http2Stream stream = stream(streamId);
-        Http2FrameType type = RST_STREAM;
-
-        if (!buffered() && hasNukleusBudget(length))
+        if (stream != null && stream.state != Http2StreamState.CLOSED)
```

Comment:
Hoist this condition into a variable, such as `needsRst`, then remove `else` clause and `return needsRst;`.